sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -219,48 +246,99 @@ private void performLogging(int logLevel, String format, Object... args) {
         // Check the configuration level every time the logger is called in case it has changed.
-        int configurationLevel =
-            Configuration.getGlobalConfiguration().get(Configuration.PROPERTY_AZURE_LOG_LEVEL, DISABLED_LEVEL);
-        if (level < configurationLevel) {
-            return false;
+        int environmentLogLevel = Configuration.getGlobalConfiguration()
+            .get(Configuration.PROPERTY_AZURE_LOG_LEVEL, DISABLED_LEVEL);
+
+        if (level < environmentLogLevel) {
+            return new LoggingConfiguration(environmentLogLevel, level, false, isExceptionLogging);
         }
 
+        boolean canLog;
         switch (level) {
             case VERBOSE_LEVEL:
-                return logger != null && logger.isDebugEnabled();
+                canLog = (logger != null && logger.isDebugEnabled());
+                break;
             case INFORMATIONAL_LEVEL:
-                return logger != null && logger.isInfoEnabled();
+                canLog = (logger != null && logger.isInfoEnabled());
+                break;
             case WARNING_LEVEL:
-                return logger != null && logger.isWarnEnabled();
+                canLog = (logger != null && logger.isWarnEnabled());
+                break;
             case ERROR_LEVEL:
-                return logger != null && logger.isErrorEnabled();
+                canLog = (logger != null && logger.isErrorEnabled());
+                break;
             default:
-                return false;
+                canLog = false;
+                break;
         }
+
+        return new LoggingConfiguration(environmentLogLevel, level, canLog, isExceptionLogging);
     }
 
     /*
-     * Removes the last element from the arguments if it is a throwable.
+     * Determines if the arguments contains a throwable that would be logged, SLF4J logs a throwable if it is the last
+     * element in the argument list.
      *
-     * @param args Arguments
-     * @return The arguments with the last element removed if it was a throwable, otherwise the unmodified arguments.
+     * @param args The arguments passed to format the log message.
+     * @return True if the last element is a throwable, false otherwise.
      */
-    private Object[] attemptToRemoveThrowable(Object... args) {
+    private boolean doesArgsHaveThrowable(Object... args) {
         if (args.length == 0) {
-            return args;
+            return false;
+        }
+
+        return args[args.length - 1] instanceof Throwable;
+    }
+
+    /*
+     * Removes the last element from the arguments as it is a throwable.
+     *
+     * @param args The arguments passed to format the log message.
+     * @return The arguments with the last element removed.
+     */
+    private Object[] removeThrowable(Object... args) {
+        return Arrays.copyOf(args, args.length - 1);
+    }
+
+    /*
+     * Helper class which contains logging configuration.
+     */
+    private static class LoggingConfiguration {
```

Comment:
I think that too many of these `LoggingConfiguration` objects will be created even when logging is not enabled. Is there a way to check `canLog` condition first and then proceed with creating this object?

Reply:
Removed `LoggingConfiguration` class.